### ISSUE TITLE
Handle reboot failures

### ISSUE THREAD

**Issue by lukaszachy** [teemtee/tmt#1405]
Currently `reboot()` return `bool` indicating if reboot happened or not.
However this value is ignored by execute (https://github.com/teemtee/tmt/blob/main/tmt/steps/execute/internal.py#L262)

IMO we should raise exception instead and handle correctly (stop execution) if reboot fails to happen (e.g. fail to boot, reboot ignored etc)

**Comment by idorax** [comment]
To reproduce `reboot` ,  we can use provision `connect`, 
e.g.
```bash
(tmt) huanli@FOO:plans$ cat reboot.fmf 
summary: Basic smoke test
execute:
    how: tmt
    script: uname -a
(tmt) huanli@FOO:plans$ 
(tmt) huanli@FOO:plans$ tmt run -a provision --how=connect --guest=kvm-01-guest01 reboot
/var/tmp/tmt/run-001

/plans/reboot
    discover
        how: shell
        summary: 1 test selected
    provision
        how: connect
        guest: kvm-01-guest01.lab.eng.brq2.redhat.com
        user: root
        key: not provided
        summary: 1 guest provisioned
    prepare
        summary: 0 preparations applied
    execute
        how: tmt
        summary: 1 test executed  
    report
        how: display
        summary: 1 test passed
    finish
        reboot: Rebooting guest
        reboot: Reboot finished
        summary: 1 task completed

total: 1 test passed

(tmt) huanli@FOO:plans$ cat /var/tmp/tmt/run-001/log.txt
...<snip>...
14:21:19         Execute command 'reboot' on guest 'kvm-01-guest01'.
14:21:19         Run command 'ssh -oForwardX11=no -oStrictHostKeyChecking=no -oUserKnownHostsFile=/dev/null -oIdentitiesOnly=yes -iNone -S/run/user/1001/tmt/tmpfutsubjj root@kvm-01-guest01 export TMT_PLAN_DATA=/var/tmp/tmt/run-005/plans/reboot/data; reboot'.
#^^^^^^^^^^^^^^^^^^^^^^^^ reboot is done via ssh
```

**Comment by idorax** [comment]
@lukaszachy, I failed to trigger off the code path `ExecuteInternal._handle_reboot()` via `tmt run -a provision --how=... reboot`, would you please let me know how can I make `tmt run` trigger off the related code path?
* https://github.com/teemtee/tmt/blob/main/tmt/steps/execute/internal.py#L214
  ```python
   187	    def _handle_reboot(self, test, guest):
   188	        """
   189	        Reboot the guest if the test requested it.
   190	
   191	        Check for presence of a file signalling reboot request
   192	        and orchestrate the reboot if it was requested. Also increment
   193	        REBOOTCOUNT variable, reset it to 0 if no reboot was requested
   194	        (going forward to the next test). Return whether reboot was done.
   195	        """
   196	        if self._will_reboot(test):
   ...
   212	            os.remove(reboot_request_path)
   213	            guest.push(test_data)
   214	            try:
   215	                guest.reboot(command=reboot_command, timeout=timeout)
   216	            except tmt.utils.RunError:
   217	                self.fail(
   218	                    f"Failed to reboot guest using the "
   219	                    f"custom command '{reboot_command}'.")
   220	                raise
   221	            except tmt.utils.ProvisionError:
   222	                self.warn(
   223	                    "Guest does not support soft reboot, "
   224	                    "trying hard reboot.")
   225	                guest.reboot(hard=True, timeout=timeout)
   226	            return True
   227	        return False
  ```

**Comment by lukaszachy** [comment]
@idorax Execute 'tmt-reboot' within the test. Inspiration in any test which uses this command.

**Comment by FrNecas** [comment]
Isn't this solved by #1487 ? I just didn't realize that there is a duplicate

**Comment by idorax** [comment]
Well, @lukaszachy and @FrNecas, can I just close this issue now?

**Comment by lukaszachy** [comment]
I'd say @FrNecas should make his PR fix this issue as well. (if it really fixes it ;)

**Comment by idorax** [comment]
Thanks @lukaszachy, I updated the assignee to @FrNecas just now.

**Comment by FrNecas** [comment]
Fixed in #1487
